### PR TITLE
A19 tasks are implemented

### PR DESCRIPTION
**Pull request by rajeevramanTMM**

### CODE REVIEW

File: Parser/Node.cs [hunk 1]
```diff
@@ -48,6 +59,30 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
...
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NCallStmt (Token Name, NExpr[] Expr) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
+
+public record NIfStmt (NExpr Expr, NStmt[] Stmts) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
```
Comment (tarydon) [inline Parser/Node.cs]: Change this to:
`public record NIfStmt (NExpr Condition, NStmt IfPart, NStmt? ElsePart)`

File: Parser/Node.cs [hunk 2]
```diff
@@ -48,6 +59,30 @@ public record NWriteStmt (bool NewLine, NExpr[] Exprs) : NStmt {
...
+
+public record NWhileStmt (NExpr Expr, NStmt[] Stmts) : NStmt {
+   public override T Accept<T> (Visitor<T> visitor) => visitor.Visit (this);
+}
```
Comment (tarydon) [inline Parser/Node.cs]: The body of a while loop is a _single_ statement. Even if there are multiple operations to complete, it is still a single CompoundStmt. 
`public record NWhileStmt (NExpr Condition, NStmt Body)`

File: Parser/Parser.cs [hunk 3]
```diff
@@ -67,11 +81,57 @@ NBlock Block ()
...
+   NCallStmt CallStmt () => new (Prev, ArgList ());
+
+   NForStmt ForStmt () {
+      var token = Expect (IDENT, "Expecting identifier");
+      Expect (ASSIGN);
+      var expr = Expression ();
+      bool to = Match (TO);
+      if (!to) Expect (DOWNTO);
+      var expr1 = Expression ();
+      Expect (DO, "Expecting do statement");
```
Comment (tarydon) [inline Parser/Parser.cs]: It's just a **DO** keyword not a **do statement**

**Review by tarydon — CHANGES_REQUESTED** [review]

**Comment by rajeevramanTMM** [comment]
Review changes applied

**Pull request closed without merging**